exporter/exporterhelper/batch_sender.go

```diff
@@ -136,7 +137,7 @@ func (bs *batchSender) resetTimer() {
 // Caller must hold the lock.
 func (bs *batchSender) isActiveBatchReady() bool {
 	return bs.activeBatch.request.ItemsCount() >= bs.cfg.MinSizeItems ||
-		(bs.concurrencyLimit > 0 && bs.activeRequests.Load() >= bs.concurrencyLimit)
+		(bs.concurrencyLimit > 0 && bs.activeBatch.goroutines >= bs.concurrencyLimit)
```

Comment:
concurrencyLimit is the limit of incoming requests = number of queue workers. The number of spawn goroutines is unbound. The reason is to avoid blocking the exporter pipeline where queue and batching both enabled until flush timeout